pkg/sinks/prometheus.go

```diff
@@ -0,0 +1,104 @@
+package sinks
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/utils/strings/slices"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/resmoio/kubernetes-event-exporter/pkg/kube"
+	"github.com/rs/zerolog/log"
+)
+
+func newGaugeVec(opts prometheus.GaugeOpts, labelNames []string) *prometheus.GaugeVec {
+	v := prometheus.NewGaugeVec(opts, labelNames)
+	prometheus.MustRegister(v)
+	return v
+}
+
+type PrometheusConfig struct {
+	EventsMetricsNamePrefix string              `yaml:"eventsMetricsNamePrefix"`
+	ReasonFilter            map[string][]string `yaml:"reasonFilter"`
+}
+
+type PrometheusGaugeVec interface {
+	With(labels prometheus.Labels) prometheus.Gauge
+	Delete(labels prometheus.Labels) bool
+}
+
+type PrometheusSink struct {
+	cfg           *PrometheusConfig
+	kinds         []string
+	metricsByKind map[string]PrometheusGaugeVec
+}
+
+func NewPrometheusSink(config *PrometheusConfig) (Sink, error) {
+	if config.EventsMetricsNamePrefix == "" {
+		config.EventsMetricsNamePrefix = "event_exporter_"
+	}
+
+	metricsByKind := map[string]PrometheusGaugeVec{}
+
+	log.Info().Msgf("Initializing new Prometheus sink...")
+	kinds := []string{}
+	for kind := range config.ReasonFilter {
+		kinds = append(kinds, kind)
+		metricName := config.EventsMetricsNamePrefix + strings.ToLower(kind) + "_event_count"
+		metricLabels := []string{strings.ToLower(kind), "namespace", "reason"}
+		metricsByKind[kind] = newGaugeVec(
+			prometheus.GaugeOpts{
+				Name: metricName,
+				Help: "Event counts for " + kind + " resources.",
+			}, metricLabels)
+
+		log.Info().Msgf("Created metric: %s, will emit events: %v with additional labels: %v", kind, config.ReasonFilter[kind], metricLabels)
+	}
+
+	return &PrometheusSink{
+		cfg:           config,
+		kinds:         kinds,
+		metricsByKind: metricsByKind,
+	}, nil
+}
+
+func (o *PrometheusSink) Send(ctx context.Context, ev *kube.EnhancedEvent) error {
```

Comment:
```suggestion
func (o *PrometheusSink) Send(_ context.Context, ev *kube.EnhancedEvent) error {
```

I think context is not used anywhere in this method

Reply:
Fixed ✔️